### PR TITLE
TPC-C Preparation Part 2

### PR DESCRIPTION
**Pull request by mrks**
I realized that we will need support for unsuccessful transactions (i.e., transaction aborts). As #1759 will change the JSON result format and this will change it again, I'd rather have only a single master commit.

Also makes the inclusion of SQL metrics optional, greatly reducing the size of the resulting JSON.

To make the review easier, this is a PR into #1759.

Sorry for making this slightly confusing. Hopefully, this makes the final TPC-C PR a bit easier to read.

### CODE REVIEW

File: scripts/compare_benchmarks.py [hunk 1]
```diff
@@ -95,7 +97,28 @@ def calculate_and_format_p_value(old, new):
...
     p_value_formatted = calculate_and_format_p_value(old, new)
 
-    table_data.append([name, str(old['items_per_second']), str(len(old['runs'])), str(new['items_per_second']), str(len(new['runs'])), diff_formatted, p_value_formatted])
+    table_data.append([name, str(old['items_per_second']), str(len(old['successful_runs'])), str(new['items_per_second']), str(len(new['successful_runs'])), diff_formatted, p_value_formatted])
+
+    if (len(old['unsuccessful_runs']) > 0 or len(new['unsuccessful_runs']) > 0):
+        old_unsuccessful_per_second = float(len(old['unsuccessful_runs'])) / (old['duration'] / 1e9)
+        new_unsuccessful_per_second = float(len(new['unsuccessful_runs'])) / (new['duration'] / 1e9)
+
+        if len(old['unsuccessful_runs']) > 0.0:
```
Comment (Bensk1) [inline scripts/compare_benchmarks.py]: Is the `len` comparison with a float intended?

File: src/benchmarklib/benchmark_runner.cpp [hunk 2]
```diff
@@ -177,6 +180,10 @@ void BenchmarkRunner::_benchmark_shuffled() {
   }
   _state.set_done();
 
+  for (auto& result : _results) {
```
Comment (Bensk1) [inline src/benchmarklib/benchmark_runner.cpp]: Maybe another comment or a pointer to the comment in benchmark_item_result.hpp might be helpful for the reader.

File: scripts/compare_benchmarks.py (hunk 1 above)
Comment (mrks) [reply scripts/compare_benchmarks.py]: Nope, just copypasta.

File: src/benchmarklib/benchmark_runner.cpp (hunk 2 above)
Comment (mrks) [reply src/benchmarklib/benchmark_runner.cpp]: laik dis?

Comment (Bensk1) [reply src/benchmarklib/benchmark_runner.cpp]: Naitz

**Pull request merged**